src/rules/no-unused-modules.js

```diff
@@ -0,0 +1,714 @@
+      
+      reexports.forEach((value, key) => {
+        if (key === DEFAULT) {
+          exports.set(IMPORT_DEFAULT_SPECIFIER, { whereUsed: new Set() })
+        } else {
+          exports.set(key, { whereUsed: new Set() })
+        }
+        const reexport =  value.getImport()
+        if (!reexport) {
+          return
+        }
+        let localImport = imports.get(reexport.path)
+        let currentValue
+        if (value.local === DEFAULT) {
+          currentValue = IMPORT_DEFAULT_SPECIFIER
+        } else {
+          currentValue = value.local
+        }
+        if (typeof localImport !== 'undefined') {
+          localImport = new Set([...localImport, currentValue])
+        } else {
+          localImport = new Set([currentValue])
+        }
+        imports.set(reexport.path, localImport)
+      })
+
+      localImportList.forEach((value, key) => {
+        if (isNodeModule(key)) {
+          return
+        }
+        imports.set(key, value.importedSpecifiers)
+      })
+      importList.set(file, imports)
+      
+      // build up export list only, if file is not ignored
+      if (ignoredFiles.has(file)) {
+        return
+      }
+      namespace.forEach((value, key) => {
+        if (key === DEFAULT) {
+          exports.set(IMPORT_DEFAULT_SPECIFIER, { whereUsed: new Set() })
+        } else {
+          exports.set(key, { whereUsed: new Set() })
+        }
+      })
+    }
+    exports.set(EXPORT_ALL_DECLARATION, { whereUsed: new Set() })
+    exports.set(IMPORT_NAMESPACE_SPECIFIER, { whereUsed: new Set() })
+    exportList.set(file, exports)
+  })
+  exportAll.forEach((value, key) => {
+    value.forEach(val => {
+      const currentExports = exportList.get(val)
+      const currentExport = currentExports.get(EXPORT_ALL_DECLARATION)
+      currentExport.whereUsed.add(key)
+    })
+  })
+}
+
+/**
+ * traverse through all imports and add the respective path to the whereUsed-list 
+ * of the corresponding export
+ */
+const determineUsage = () => {
+  importList.forEach((listValue, listKey) => {
+    listValue.forEach((value, key) => {
+      const exports = exportList.get(key)
+      if (typeof exports !== 'undefined') {
+        value.forEach(currentImport => {
+          let specifier
+          if (currentImport === IMPORT_NAMESPACE_SPECIFIER) {
+            specifier = IMPORT_NAMESPACE_SPECIFIER
+          } else if (currentImport === IMPORT_DEFAULT_SPECIFIER) {
+            specifier = IMPORT_DEFAULT_SPECIFIER
+          } else {
+            specifier = currentImport
+          }
+          if (typeof specifier !== 'undefined') {
+            const exportStatement = exports.get(specifier)
+            if (typeof exportStatement !== 'undefined') {
+              const { whereUsed } = exportStatement
+              whereUsed.add(listKey)
+              exports.set(specifier, { whereUsed })
+            }
+          }
+        })
+      }
+    })
+  })
+}
+
+const getSrc = src => {
+  if (src) {
+    return src
```

Comment:
Should this function be ensuring it’s an array? If so:
```suggestion
     return [].concat(src);
```

Reply:
The json schema already ensures that the ```src``` passed to the rule config is an array.

This function should return an array containing ```process.cwd()``` in case no ```src``` has been defined within the rule config.

Reply:
Makes sense